transmit-lib/src/cmr/transmit/search.clj

```diff
@@ -51,13 +51,17 @@
   [context params]
   (let [conn (config/context->app-connection context :search)
         request-url (str (conn/root-url conn) "/granules.xml")
+        request-body (dissoc params :token)
+        token (:token params)
+        header (ch/context->http-headers context)
         response (client/post request-url
                               (merge
                                 (config/conn-params conn)
-                                {:body (codec/form-encode params)
+                                {:body (codec/form-encode request-body)
                                  :content-type mt/form-url-encoded
                                  :throw-exceptions false
-                                 :headers (ch/context->http-headers context)}))
+                                 :headers (if (nil? token) header
```

Comment:
change to `(if token (assoc header...) header)`

Reply:
that looks a lot cleaner then what I tried.